packages/wp-story-editor/src/components/autoSaveCheck/index.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useEffect, useState, useCallback } from '@googleforcreators/react';
+import { useConfig } from '@googleforcreators/story-editor';
+
+/**
+ * Internal dependencies
+ */
+import { AutoSaveDialog } from './autoSaveDialog';
+
+export const AutoSaveCheck = () => {
+  const { autoSaveLink } = useConfig();
+  const [showDialog, setShowDialog] = useState(false);
+  const onClose = () => setShowDialog(false);
+
+  const onPrimary = useCallback(() => {
+    window.location = autoSaveLink;
+  }, [autoSaveLink]);
+
+  useEffect(
+    () => {
+      if (autoSaveLink) {
+        setShowDialog(true);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- we only want to run this once
+    []
+  );
```

Comment:
I don't think we need state & an effect here.

We can just do `const isOpen = Boolean(autoSaveLink)`